src/components/ContentDetails/ContentDetails.component.jsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import htmlParser from 'html-react-parser';
+import {
+  Favorite as FavoriteIcon,
+  FavoriteBorder as FavoriteBorderIcon,
+} from '@material-ui/icons';
+
+import {
+  StyledFlexDiv,
+  StyledVideoDiv,
+  StyledRelatedDiv,
+  StyledIFrame,
+  StyledSnippetDiv,
+  StyledSnippetTitleDiv,
+  SnippetTitle,
+  SnippetDescription,
+  StyledBackButton,
+  StyledButton,
+} from './ContentDetails.styles';
+import RelatedItem from '../RelatedItem';
+
+import { useAuth } from '../../providers/Auth';
+import { useCustom } from '../../providers/Custom';
+
+import { getYoutubeEmbedLink } from '../../utils/fns';
+
+const ContentDetails = ({ item, relatedItems }) => {
+  const history = useHistory();
+  const location = useLocation();
+  const { authenticated } = useAuth();
+  const { findFavoriteVideo, addFavoriteVideo, removeFavoriteVideo } = useCustom();
+
+  const isVideoInList = Boolean(findFavoriteVideo(item.id.videoId));
+  const isFavoritePage = location.pathname.includes('favs');
+
+  const path = isFavoritePage ? 'favs' : '/';
+  const handleGoBack = () => history.push(path);
+  const handleFavButton = isVideoInList
+    ? () => removeFavoriteVideo(item)
+    : () => addFavoriteVideo(item);
+
+  return (
+    <StyledFlexDiv>
+      <StyledVideoDiv>
+        <StyledIFrame
+          title={item.snippet.title}
+          src={getYoutubeEmbedLink(item.id.videoId)}
+        />
+        <StyledSnippetDiv>
+          <StyledSnippetTitleDiv>
+            <SnippetTitle>{htmlParser(item.snippet.title)}</SnippetTitle>
+            {authenticated && !isFavoritePage && (
```

Comment:
I think that it'll be useful to have the "Remove from favorites" button on the favorites page too:

```suggestion
            {authenticated && (
```

Reply:
Sure thing! I'll commit it